source/docs/advanced/features/external-logins.md

```diff
@@ -0,0 +1,314 @@
+        }
+      }
+    )
+  );
+
+  passportAuthenticator.authenticate("facebook", {
+    failureRedirect: `${this.shopUrl}/login`,
+    session: false,
+    scope: ["email"],
+  })(req, res, next);
+}
+```
+
+### Creating your own ExternalLoginHandler
+
+The `ExternalLoginHandler` is responsible of actually logging the user into your platform. It should first create the user if he does not exist. Next it should log in the user (old or new) into the system.
+
+The external loging handler should have a method called `login` (see signature below) that when called will ensure the user exist or otherwise create him and finally log him/her in.
+
+```js
+class CustomExternalLoginHander extends ExternalLoginHandlerInterface {
+  async login(request, user) {
+    // ensure/create then login user
+  }
+}
+```
+
+We also provide a `ExternalLoginHandlerBase` class that implements the `ExternalLoginHandlerInterface`. It only implements a `constructor` with the folowing signature:
+
+```js
+constructor(globalConfig, staticConfigFromProviders) {
+}
+```
+
+`globalConfig`, `staticConfigFromProviders` are used since they are what is sent to the `makeExternalLoginEndpoint` (see below), and for the `globalConfig` you can use it as follows:
+
+```js
+import ExternalLoginHandlerBase from "auth-external-login/server/ExternalLoginHandlerBase";
+import makeLoadersFromRequest from "server/core/graphql/makeLoadersFromRequest";
+
+export default class CustomLoginHandler extends ExternalLoginHandlerBase {
+  async login(req, user) {
+    const loaders = makeLoadersFromRequest(this.globalConfig, req);
+    // you have access to all your loaders!!! like loaders.Product, loaders.Customer...
+  }
+}
+
+```
+
+### Putting it all together
+
+As discussed previously the providers provide the middleware to handle the login with the external system, and the external login handler handles actually logging they user in. Now we are going to link the providers with the external login handler in module that we can register into `.front-commerce.js`
+
+A Front-Commerce module is basically a directory that follows certain conventions. It can be placed any where in your folder structure but is commonly placed either under `./modules/your_custom_module_name` or `./src/modules/your_custom_module_name`.
+
+For our purposes we are only interensted in the server part of the module so we need to create `your_custom_module_name/server/module.config.js` file.
+
+The `your_custom_module_name/server/module.config.js` file should export an object with the providers middleware as follows:
+
+```js
+import makeExternalLoginEndpoint from "auth-external-login/server/makeExternalLoginEndpoint";
+import CustomExternalLoginHandler from "you_external_login_handler/CustomExternalLoginHandler";
+
+export default {
+  endpoint: makeExternalLoginEndpoint((config, staticConfigFromProviders) => new CustomExternalLoginHandler(/** some parameters you extract from the configs */)),
```

Comment:
long lines make code snippet hard to read

![image](https://user-images.githubusercontent.com/305563/154314686-9e23bfd7-5b1a-4f63-b0f1-375d992988fa.png)